src/tools/testwrap

```diff
@@ -1,25 +1,33 @@
-#!/bin/sh
-#
-# FIXME: I should probably be a perl or python script
-#
+#!/usr/bin/env python3
 
-# FIXME: argument parsing
+import argparse
+import re
+import shutil
+import subprocess
+import os
 
-srcdir=$1
-basedir=$2
-builddir=$3
-testgroup=$4
-testname=$(echo "$5"|sed -e 's/.pl$//')
-shift 5
+parser = argparse.ArgumentParser()
 
-testdir="$basedir/testrun/$testgroup/$testname"
-echo "# executing test in $testdir group $testgroup test $testname, builddir $builddir"
-rm -rf "$testdir/"
-mkdir -p "$testdir"
+parser.add_argument('--srcdir', help='source directory of test', type=str)
+parser.add_argument('--basedir', help='base directory of test', type=str)
+parser.add_argument('--builddir', help='build directory of meson build', type=str)
+parser.add_argument('--testgroup', help='test group', type=str)
+parser.add_argument('--test_name', help='test name', type=str)
 
-cd "$srcdir"
+args, test_command = parser.parse_known_args()
 
-export TESTOUTDIR="$testdir"
-export TESTDIR="$builddir"
+args.test_name = re.sub('.pl$', '', args.test_name)
 
-exec "$@"
+testdir = '{}/testrun/{}/{}'.format(args.basedir, args.testgroup, args.test_name)
+
+print('# executing test in {} group {} test {}, builddir {}'
+  .format(testdir, args.testgroup, args.test_name, args.builddir))
+
+if os.path.exists(testdir) and os.path.isdir(testdir):
+  shutil.rmtree(testdir)
+os.makedirs(testdir)
+
+os.chdir(args.srcdir)
+
+env_dict = dict(TESTOUTDIR=testdir, TESTDIR=args.builddir, **os.environ)
```

Comment:
It's much more common to initialize dicts with {} in python... And it seems not quite right to allow os.environ to overwrite TESTOUTDIR etc?